ui/src/overview.tsx

```diff
@@ -298,9 +383,13 @@ function ParticipantCriteria(props: {
               {props.criteria.name}
             </Link>
             <Divider orientation="vertical" variant="middle" flexItem />
-            <Typography variant="body1">
-              {0 /* TODO(tjennison): Fetch from backend. */}
-            </Typography>
+            {
+              <Loading status={criteriaCountState}>
```

Comment:
Does this display a sensibly sized spinner and not a giant one like elsewhere in the UI?

Reply:
Would it make sense to modify the existing `Loading` component to have an optional `size` prop, so smaller spinners can be specified? Or would you prefer me to just render a Material UI spinner directly in these cases?

Reply:
You should edit the `Loading` component so the handling is consistent. However, it's also needs to consider what happens with errors in the small space. Look at what the `TreeGrid` does, using an icon for errors that can be clicked to reload and showing the error message as a mouse over.

Reply:
I've modified the `Loading` component to render customized spinners and error messages if a `size` prop with value `small` is supplied. The spinners and error messages are displaying appropriately on the UI, but let me know if you have any concerns about the implementation.